Build failure: gnuplot can't build with TeXLive

### Steps To Reproduce

Steps to reproduce the behavior:
1. build gnuplot with TeXLive

### Build log

```
       > make[2]: Entering directory '/build/gnuplot-6.0.1/share/LaTeX'
       > make[3]: Entering directory '/build/gnuplot-6.0.1/share/LaTeX'
       > make[3]: Nothing to be done for 'install-exec-am'.
       >  /nix/store/ph44jcx3ddmlwh394mh1wb7f1qigxqb1-coreutils-9.5/bin/mkdir -p '/nix/store/vinrz886mrmiigp9qssxr0rbc8qzl3a8-texlive-2023-env/share/texmf-local/tex/latex/gnuplot'
       > /nix/store/ph44jcx3ddmlwh394mh1wb7f1qigxqb1-coreutils-9.5/bin/mkdir: cannot create directory '/nix/store/vinrz886mrmiigp9qssxr0rbc8qzl3a8-texlive-2023-env/share/texmf-local': Permission denied
       > make[3]: *** [Makefile:348: install-texDATA] Error 1
       > make[3]: Leaving directory '/build/gnuplot-6.0.1/share/LaTeX'
       > make[2]: *** [Makefile:419: install-am] Error 2
```

### Additional context

Add any other context about the problem here.

### Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@lovek323

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.56, NixOS, 24.11 (Vicuna), 24.11.20241017.b4802f7`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.9`
 - channels(root): `"nixos"`
 - nixpkgs: `/nix/store/jqn0kbmflh5wdr0809rj7xsmbm9w1y20-source`
```

---

--- a/pkgs/tools/graphics/gnuplot/default.nix
+++ b/pkgs/tools/graphics/gnuplot/default.nix
@@ -49,7 +49,8 @@ in
     (if withX then "--with-x" else "--without-x")
     (if withQt then "--with-qt=qt5" else "--without-qt")
     (if aquaterm then "--with-aquaterm" else "--without-aquaterm")
-  ] ++ lib.optional withCaca "--with-caca";
+  ] ++ lib.optional withCaca "--with-caca"
+    ++ lib.optional withTeXLive "--with-texdir=${placeholder "out"}/share/texmf/tex/latex/gnuplot";
 
   CXXFLAGS = lib.optionalString (stdenv.hostPlatform.isDarwin && withQt) "-std=c++11";
 

